### PR TITLE
Feature/init mail integration

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Init email integration.

Requires #67 to run.
#69 provides corresponding dev stack.

### CODE REVIEW

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/application/usecase/SendMailUseCase.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            this.sendMail(mailModel, "templates/email-logo.png");
+
+        } catch (IOException ioException) {
+            throw new TemplateError("The template " + mail.getTemplate() + " could not be loaded");
+        } catch (TemplateException templateException) {
+            throw new TemplateError(templateException.getMessage());
+        }
+    }
+
+    private de.muenchen.refarch.email.model.Mail createMail(BasicMail mail) {
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/application/usecase/SendMailUseCase.java:65]: dont use fully qualified name

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/application/usecase/SendMailUseCase.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                .receivers(mail.getReceivers())
+                .subject(mail.getSubject())
+                .replyTo(mail.getReplyTo())
+                .receiversCc(mail.getReceiversCc())
+                .receiversBcc(mail.getReceiversBcc())
+                .attachments(attachments)
+                .build();
+    }
+
+    private void sendMail(de.muenchen.refarch.email.model.Mail mailModel, String logoPath) throws MailSendException {
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/application/usecase/SendMailUseCase.java:80]: dont use fully qualified name

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TemplateMail.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@RequiredArgsConstructor
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TemplateMail.java:14]:
```suggestion

```

`@Data` already contains `RequiredArgsConstructor`

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TextMail.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Object contains all the information needed to send a mail.
+ */
+@EqualsAndHashCode(callSuper = true)
+@Data
+@RequiredArgsConstructor
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TextMail.java:15]:
```suggestion

```

`@Data` already contains `RequiredArgsConstructor`

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/test/java/de/muenchen/refarch/email/integration/adapter/out/s3/S3AdapterTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
+package de.muenchen.refarch.email.integration.adapter.out.s3;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/test/java/de/muenchen/refarch/email/integration/adapter/out/s3/S3AdapterTest.java:4]: Dont use Wildcard imports. Also remove them on other places. Try to also set it inside your IDE, so you can fix it faster and it doesnt happen again.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+
+@SpringBootApplication
+public class EmailRestExampleApplication {
+    @Autowired
+    TestService testService;
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java:12]: dont use a field injection

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static void main(final String[] args) {
+        SpringApplication.run(EmailRestExampleApplication.class, args);
+    }
+
+    @EventListener(ApplicationReadyEvent.class)
+    void sendTestMail() {
+        this.testService.testSendMail();
+        System.exit(0);
+    }
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java:22]: why do we need that?

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/BasicMail.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import jakarta.validation.constraints.NotBlank;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class BasicMail {
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/BasicMail.java:12]: can we use records for all models?

File: refarch-integrations/refarch-email-integration/refarch-email/refarch-email-starter/src/main/java/de/muenchen/refarch/email/configuration/DigiwfEmailAutoConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.web.servlet.view.freemarker.FreeMarkerConfigurer;
+
+@RequiredArgsConstructor
+@EnableConfigurationProperties({ MailProperties.class, CustomMailProperties.class })
+public class DigiwfEmailAutoConfiguration {
```
Comment (DanielOber) [inline refarch-integrations/refarch-email-integration/refarch-email/refarch-email-starter/src/main/java/de/muenchen/refarch/email/configuration/DigiwfEmailAutoConfiguration.java:20]: Remove `Digiwf` from name

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/BasicMail.java (hunk 8 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/BasicMail.java:12]: I looked into that and as far as i understood it's not possible to extends records.
So i would say no.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java (hunk 7 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/java/de/muenchen/refarch/email/integration/EmailRestExampleApplication.java:22]: Just for having a runnable example and for e2e testing.
Could theoretically be done with testcontainers but with this there is also a real usage example.
Would let it this way for the moment.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TextMail.java (hunk 4 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TextMail.java:15]: For some reason that breaks the build, but i made the classes unmodifiable and removed the `@Data` annotation.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TemplateMail.java (hunk 3 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-core/src/main/java/de/muenchen/refarch/email/integration/domain/model/TemplateMail.java:14]: See other

**Review by DanielOber — APPROVED** [review]
LGTM

**Pull request merged**